extensions/git/src/git.ts

```diff
@@ -1993,6 +1998,24 @@ export class Repository {
 			child.on('error', cpErrorHandler(e));
 			child.on('exit', onExit);
 		});
+
+		if (opts?.cancellationToken) {
+			const cancellationPromise = new Promise<{ status: IFileStatus[]; statusLength: number; didHitLimit: boolean }>((_, e) => {
+				onceEvent(opts.cancellationToken!.onCancellationRequested)(() => {
+					try {
+						child.kill();
+					} catch (err) {
+						// noop
+					}
+
+					e(new GitError({ message: 'Cancelled' }));
```

Comment:
Should this rather be a `new CancellationError` instead? We might have to create such a concept.